docs/usage/self-hosted-configuration.md

```diff
@@ -324,31 +324,92 @@ This private key is used to decrypt config files.
+Requested keysize is 4096 bits
+Please specify how long the key should be valid.
+         0 = key does not expire
+      <n>  = key expires in n days
+      <n>w = key expires in n weeks
+      <n>m = key expires in n months
+      <n>y = key expires in n years
+Key is valid for? (0)
+Key does not expire at all
+Is this correct? (y/N) y
+
+GnuPG needs to construct a user ID to identify your key.
+
+Real name: Renovate Bot
+Email address: renovate@whitesourcesoftware.com
+Comment:
+You selected this USER-ID:
+    "Renovate Bot <renovate@whitesourcesoftware.com>"
+
+Change (N)ame, (C)omment, (E)mail or (O)kay/(Q)uit? O
+
+gpg: key 0649CC3899F22A66 marked as ultimately trusted
+gpg: revocation certificate stored as '/Users/rhys/.gnupg/openpgp-revocs.d/794B820F34B34A8DF32AADB20649CC3899F22A66.rev'
+public and secret key created and signed.
+
+pub   rsa4096 2021-09-10 [SC]
+      794B820F34B34A8DF32AADB20649CEXAMPLEONLY
+uid                      Renovate Bot <renovate@whitesourcesoftware.com>
+sub   rsa4096 2021-09-10 [E]
+```
+
+</details>
+
+- Copy the key ID from the output (`794B820F34B34A8DF32AADB20649CEXAMPLEONLY` in the above example) or run `gpg --list-secret-keys` if you forgot to take a copy
+- Run `gpg --armor --export-secret-keys YOUR_NEW_KEY_ID > renovate-private-key.asc` to generate an armored (text-based) private key file
+- Run `gpg --armor --export YOUR_NEW_KEY_ID > renovate-public-key.asc` to generate an armored (text-based) public key file
+
+The private key should then be added to your Renovate Bot global config (either using `privateKeyPath` or exporting it to the `RENOVATE_PRIVATE_KEY` environment variable).
+The public key can be used to replace the existing key in <https://renovatebot.com/encrypt> for your own use.
 
-Replace `actual-secret` with the secret to encrypt.
+Any encrypted secrets using GPG must have a mandatory organization/group scope, and optionally can be scoped for a single repository only.
+The reason for this is to avoid "replay" attacks where someone could learn your encrypted secret and then reuse it in their own Renovate repositories.
+Instead, with scoped secrets it means that Renovate ensures that the organization and optionally repository values encrypted with the secret match against the running repository.
+
+Note: simple public key encryption was previously used to encrypt secrets, but this approach has now been deprecated and no longer documented.
 
 ## privateKeyOld
 
 Use this field if you need to perform a "key rotation" and support more than one keypair at a time.
 Decryption with this key will be attempted after `privateKey`.
 
+If you are migrating from the legacy public key encryption approach to use GPG, then move your legacy private key from `privateKey` to `privateKeyOld` and then put your new GPG private key in `privateKey`.
+Doing so will mean that Renovate will first attempt to decrypt using the GPG key but fall back to the legacy key and try that next.
```

Comment:
I'm missing an explanation on when it's safe to remove the `privateKeyOld` config option when doing the "key rotation".

Maybe we want a user to do something like this?

1. Create new GPG key.
1. Put old key in `privateKeyOld` config option.
1. Confirm new GPG key works (how would you even confirm this if the old key keeps hanging around, though...)
1. Maybe wait until the PRs which use `privateOldKey` have been merged???
1. Remove `privateKeyOld` config option.

I'll let you respond and decide if this needs to go in the text. 😉

Reply:
They can remove it once all the old encrypted values have been migrated, or they wish to no longer support the old key and let repos fail